src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java

```diff
@@ -61,53 +61,64 @@ public class TaxonomyServiceImpl implements WfServiceHandler {
-            List <HashMap<String, Object>> updateFieldValues = null;
-            if (state.equals(Constants.INITIATE)){
-                requestMap.put(Constants.APPROVAL_STATUS, draft);
-            } else if (state.equals(Constants.SEND_FOR_REVIEW_LEVEL_1)){
-                requestMap.put(Constants.APPROVAL_STATUS, under_L1_Review);
-            } else if (state.equals(Constants.SEND_FOR_REVIEW_LEVEL_2)){
-                requestMap.put(Constants.APPROVAL_STATUS, under_L2_Review);
-            } else if (state.equals(Constants.APPROVED)){
-                requestMap.put(Constants.APPROVAL_STATUS, approved);
-            }
+    @Value("{$taxonomy.term.read.api}")
+    private String termReadURI;
 
-            if (state.equals(Constants.INITIATE)){
-                updateFieldValues  =  wfRequest.getUpdateFieldValues();
-            } else {
-                WfStatusEntityV2 wfStatusEntityV2 =  wfStatusRepoV2.findBywfId(wfRequest.getWfId());
-                updateFieldValues = getUpdateFieldValues(wfStatusEntityV2);
-            }
+    @Value("{$taxonomy.workflow.live}")
+    private String live;
 
+    @Value("{$taxonomy.workflow.publish}")
+    private String underPublish;
+    @Override
+    public void processMessage(WfRequest wfRequest) {
+        if (Objects.nonNull(wfRequest) && !StringUtils.isEmpty(wfRequest.getWfId())) {
+            WfStatusEntityV2 wfStatusEntityV2 = wfStatusRepoV2.findBywfId(wfRequest.getWfId());
+            List<HashMap<String, Object>> updateFieldValues = getUpdateFieldValues(wfStatusEntityV2);
             if (Objects.nonNull(updateFieldValues) && !CollectionUtils.isEmpty(updateFieldValues)) {
                 for (HashMap<String, Object> updateFieldValue : updateFieldValues) {
                     String identifier = (String) updateFieldValue.get(Constants.IDENTIFIER);
                     String id = (String) updateFieldValue.get(Constants.CODE);
                     String category = (String) updateFieldValue.get(Constants.CATEGORY);
-                    requestMap.put(Constants.IDENTIFIER, identifier);
-                    term.put(Constants.TERM,requestMap);
-                    request.put(Constants.REQUEST,term);
-                    String URI = constructTermUpdateURI(id, category);
-                    logger.info("printing URI For Term Update {} ", URI);
-                    Map<String, Object> response = requestService.fetchResultUsingPatch(URI,request, null);
-                    String responseCode = (String) response.get(Constants.RESPONSE_CODE);
-                    if (!responseCode.equals(Constants.OK)){
-                        logger.error("Unable To Update Term Status");
+                    StringBuilder termRead = constructReadTermApiURI(id, category);
+                    Map<String, Object> termResponse = (Map<String, Object>) requestService.fetchResultUsingGet(termRead);
+                    String responseForTermRead = (String) termResponse.get(Constants.RESPONSE_CODE);
```

Comment:
Do not create variables unnecessarily... you can write the following... 
if (Constants.OK.equalsIgnoreCase((String) termResponse.get(Constants.RESPONSE_CODE)))